components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/service/WebSubHubAdapterServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.identity.event.common.publisher.model.EventContext;
+import org.wso2.identity.event.common.publisher.EventPublisher;
+import org.wso2.identity.event.websubhub.publisher.internal.WebSubHubAdapterDataHolder;
+import org.wso2.identity.event.common.publisher.model.SecurityEventTokenPayload;
+import org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants;
+import org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil;
+import org.wso2.identity.event.websubhub.publisher.exception.WebSubAdapterException;
+
+import java.io.IOException;
+
+import static org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil.handleServerException;
+
+/**
+ * OSGi service for publishing events using web sub hub.
+ */
+public class WebSubHubAdapterServiceImpl implements EventPublisher {
+
+    private static final Log log = LogFactory.getLog(WebSubHubAdapterServiceImpl.class);
+    private String webSubHubBaseUrl = null;
+
+    @Override
+    public void publish(SecurityEventTokenPayload eventPayload, EventContext eventContext, String eventUri)
+            throws WebSubAdapterException {
+
+        log.info("Event published with new generic component.");
+        if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().isAdapterEnabled()) {
+            WebSubHubAdapterUtil.makeAsyncAPICall(eventPayload, eventContext, constructHubTopic(eventUri,
+                            eventContext.getTenantDomain()),
+                    getWebSubBaseURL());
+        } else {
+            log.warn("Event cannot be published, WebSub Hub Adapter is not enabled.");
+            throw WebSubHubAdapterUtil.handleClientException(WebSubHubAdapterConstants.ErrorMessages.WEB_SUB_HUB_ADAPTER_DISABLED);
+        }
+    }
+
+    public void registerTopic(String eventUri, String tenantDomain) throws WebSubAdapterException {
+
+        if (WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().isAdapterEnabled()) {
+            try {
+                WebSubHubAdapterUtil.makeTopicMgtAPICall(constructHubTopic(eventUri, tenantDomain), getWebSubBaseURL(), WebSubHubAdapterConstants.REGISTER);
+            } catch (IOException e) {
+                throw WebSubHubAdapterUtil.handleServerException(WebSubHubAdapterConstants.ErrorMessages.ERROR_REGISTERING_HUB_TOPIC, e, eventUri, tenantDomain);
+            }
+        } else {
+            log.warn("WebSub Hub Topic cannot be registered, WebSub Hub Adapter is not enabled.");
```

Comment:
IMO if adapter is not enabled we don't need to this implementation to identify it's not enabled.
From the publishing core implementation we should be able to invoke adopters only enabled.
Having said that enabling a publisher is not a runtime config but defined at server level which needs a restart we don't need this check at all

Reply:
Fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/eb4cae044fe65111cb72917fd78af0619d056a02